parts/xmem_nozzle.go

```diff
@@ -1779,18 +1779,22 @@ func (xmem *XmemNozzle) batchGet(get_map base.McRequestMap, getSpecWithHlv, getS
 				noRep_map[uniqueKey] = RetryTargetLocked
 				delete(get_map, uniqueKey)
 			} else if ok && base.IsSuccessGetResponse(resp.Resp) {
-				var res base.ConflictResult
-				res, err = xmem.conflict_resolver(wrappedReq, resp.Resp, resp.Specs, xmem.sourceBucketId, xmem.targetBucketId, xmem.xattrEnabled, xmem.uncompressBody, xmem.Logger())
+				res, isImport, err := xmem.conflict_resolver(wrappedReq, resp.Resp, resp.Specs, xmem.sourceBucketId, xmem.targetBucketId, xmem.xattrEnabled, xmem.uncompressBody, xmem.Logger())
 				if err != nil {
 					// Log the error. We will retry
 					xmem.Logger().Errorf("%v conflict_resolver: '%v'", xmem.Id(), err)
 					continue
 				}
 				switch res {
 				case base.SendToTarget:
+					// Import mutations sent will be counted when we send since we will get a more accurate count then.
+					// If target document does not exist, we only parse for importCas at send time.
 					sendLookupMap[uniqueKey] = resp
 				case base.Skip:
 					noRep_map[uniqueKey] = NotSendFailedCR
+					if isImport {
+						xmem.RaiseEvent(common.NewEvent(common.ImportMutationsSkipped, nil, xmem, nil, nil))
```

Comment:
Why break off a new `isImport` case?
Could we have just kept using "DataFailedCRSource" event and added more information as part of the "additionalInfo"?
Then, for stats handling, we can increment the mobile stat as appropriate.

Ideally, what we should do is have a "importMutation" boolean in "WrappedMCRequest" set to true.
Then, any future stat handling, such as "sent", "notSentFailCR", "Filtered", etc, etc can just have its current handling of events but also tag increment import stats as needed.

The reason I say this is because this one-off way of doing things is more than likely to introduce leakage of stats.  For example, now we want to know if a document filtered from being replicated is a mobile import, and now we gotta go add another "if" statement, which is missing right now.

Reply:
Done